const not available in array declaration (`len`)

Tried defining a multi-dimensional array, like so:

```v
const (
	grid_size			= 2000
)

fn main() {
	mut data := [][]int{ len: grid_size, init: []int{ len: grid_size } }
}
```

And this fails to build (`v run simple.v`):

```
$ v run simple.v 
simple.v:6:6: warning: unused variable: `data`
    4 | 
    5 | fn main() {
    6 |     mut data := [][]int{ len: grid_size, init: []int{ len: grid_size } }
      |         ~~~~
    7 | }
==================
/tmp/v/simple.tmp.c: In function ‘main__main’:
/tmp/v/simple.tmp.c:9392:141: error: ‘grid_size’ undeclared (first use in this function); did you mean ‘dysize’?
  array_array_int data = __new_array_with_array_default(_const_main__grid_size, 0, sizeof(array_int), (array_int[]){__new_array_with_default(grid_size, 0, sizeof(int), 0)}[0]);
                                                                                                                                             ^~~~~~~~~
                                                                                                                                             dysize
/tmp/v/simple.tmp.c:9392:141: note: each undeclared identifier is reported only once for each function it appears in
/tmp/v/simple.tmp.c: In function ‘_vinit’:
/tmp/v/simple.tmp.c:9418:38: warning: integer constant is so large that it is unsigned
  _const_math__bits__max_u64 = ((u64)(18446744073709551615));
                                      ^~~~~~~~~~~~~~~~~~~~
/tmp/v/simple.tmp.c:9445:409: warning: integer constant is so large that it is unsigned
   ((u64)(1)), ((u64)(10)), ((u64)(100)), ((u64)(1000)), ((u64)(10000)), ((u64)(100000)), ((u64)(1000000)), ((u64)(10000000)), ((u64)(100000000)), ((u64)(1000000000)), ((u64)(10000000000)), ((u64)(100000000000)), ((u64)(1000000000000)), ((u64)(10000000000000)), ((u64)(100000000000000)), ((u64)(1000000000000000)), ((u64)(10000000000000000)), ((u64)(100000000000000000)), ((u64)(1000000000000000000)), ((u64)(10000000000000000000))}));
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

I was following the [multi-dimensional array docs](https://github.com/vlang/v/blob/master/doc/docs.md#multidimensional-arrays).

**V version:** V 0.1.29 ce4ee2b
**OS:** Ubuntu 18.04

**What did you do?**
Tried to define an array using a constant length.

**What did you expect to see?**
No error / output
 
**What did you see instead?**
A c-level crash.

--- a/vlib/builtin/array_test.v
+++ b/vlib/builtin/array_test.v
@@ -1032,3 +1032,18 @@ fn test_direct_array_access_via_ptr() {
 	}
 }
 
+const (
+	grid_size_1 = 2
+	grid_size_2 = 3
+	grid_size_3 = 4
+	cell_value = 123
+)
+
+fn test_multidimensional_array_initialization_with_consts() {
+	mut data := [][][]int{ len: grid_size_1, init: [][]int{ len: grid_size_2, init: []int{ len: grid_size_3, init: cell_value } } }
+	assert data.len == grid_size_1
+	assert data[0].len == grid_size_2
+	assert data[0][0].len == grid_size_3
+	assert data[0][0][0] == cell_value
+	assert data[1][1][1] == cell_value
+}

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -2065,6 +2065,9 @@ pub fn (mut c Checker) array_init(mut array_init ast.ArrayInit) table.Type {
 			}
 		}
 		sym := c.table.get_type_symbol(array_init.elem_type)
+		if array_init.has_default {
+			c.expr(array_init.default_expr)
+		}
 		if sym.kind == .placeholder {
 			c.error('unknown type `$sym.source_name`', array_init.elem_type_pos)
 		}

